server/src/domain/auth/auth.dto.ts

```diff
@@ -0,0 +1,132 @@
+import { UserEntity, UserTokenEntity } from '@app/infra/entities';
+import { ApiProperty } from '@nestjs/swagger';
+import { Transform } from 'class-transformer';
+import { IsEmail, IsNotEmpty, IsString, MinLength } from 'class-validator';
+
+export class AuthUserDto {
+  id!: string;
+  email!: string;
+  isAdmin!: boolean;
+  isPublicUser?: boolean;
+  sharedLinkId?: string;
+  isAllowUpload?: boolean;
+  isAllowDownload?: boolean;
+  isShowMetadata?: boolean;
+  accessTokenId?: string;
+  externalPath?: string | null;
+}
+
+export class LoginCredentialDto {
+  @IsEmail({ require_tld: false })
+  @Transform(({ value }) => value?.toLowerCase())
+  @IsNotEmpty()
+  @ApiProperty({ example: 'testuser@email.com' })
+  email!: string;
+
+  @IsString()
+  @IsNotEmpty()
+  @ApiProperty({ example: 'password' })
+  password!: string;
+}
+
+export class LoginResponseDto {
+  accessToken!: string;
+  userId!: string;
+  userEmail!: string;
+  firstName!: string;
+  lastName!: string;
+  profileImagePath!: string;
+  isAdmin!: boolean;
+  shouldChangePassword!: boolean;
+}
+
+export function mapLoginResponse(entity: UserEntity, accessToken: string): LoginResponseDto {
+  return {
+    accessToken: accessToken,
+    userId: entity.id,
+    userEmail: entity.email,
+    firstName: entity.firstName,
+    lastName: entity.lastName,
+    isAdmin: entity.isAdmin,
+    profileImagePath: entity.profileImagePath,
+    shouldChangePassword: entity.shouldChangePassword,
+  };
+}
+
+export class LogoutResponseDto {
+  successful!: boolean;
+  redirectUri!: string;
+}
+
+export class SignUpDto extends LoginCredentialDto {
+  @IsString()
+  @IsNotEmpty()
+  @ApiProperty({ example: 'Admin' })
+  firstName!: string;
+
+  @IsString()
+  @IsNotEmpty()
+  @ApiProperty({ example: 'Doe' })
+  lastName!: string;
+}
+
+export class ChangePasswordDto {
+  @IsString()
+  @IsNotEmpty()
+  @ApiProperty({ example: 'password' })
+  password!: string;
+
+  @IsString()
+  @IsNotEmpty()
+  @MinLength(8)
```

Comment:
Do we want to have a more strict password policy?

Reply:
I think it is ok for now